src/ucp/core/ucp_worker.c

```diff
@@ -1725,6 +1725,9 @@ ucp_worker_get_ep_config(ucp_worker_h worker, const ucp_ep_config_key_t *key,
 
 out:
     *cfg_index_p = ep_cfg_index;
+    if (key->err_mode != UCP_ERR_HANDLING_MODE_NONE) {
+        ucp_worker_keepalive_add_ep(worker);
+    }
```

Comment:
move this to ucp_wireup_init_lanes() and pass ep to ucp_worker_keepalive_add_ep() instead of worker. the function ucp_worker_keepalive_add_ep() will check ep_config->key.err_mode